graphrag/storage/cosmosdb_pipeline_storage.py

```diff
@@ -0,0 +1,300 @@
+                return item_json_str
+        except Exception:
+            log.exception("Error reading item %s", key)
+            return None
+        else:
+            return None
+
+    async def set(self, key: str, value: Any, encoding: str | None = None) -> None:
+        """Set a file in the database for the given key."""
+        try:
+            if self._current_container:
+                container_client = self._database_client.get_container_client(
+                    self._current_container
+                )
+                if isinstance(value, bytes):
+                    value_df = pd.read_parquet(BytesIO(value))
+                    value_json = value_df.to_json(
+                        orient="records", lines=False, force_ascii=False
+                    )
+                    if value_json is None:
+                        log.exception("Error converting output %s to json", key)
+                    else:
+                        cosmos_db_item = {"id": key, "body": json.loads(value_json)}
+                        container_client.upsert_item(body=cosmos_db_item)
+                else:
+                    cosmos_db_item = {"id": key, "body": json.loads(value)}
+                    container_client.upsert_item(body=cosmos_db_item)
+        except Exception:
+            log.exception("Error writing item %s", key)
+
+    async def has(self, key: str) -> bool:
+        """Check if the given file exists in the cosmosdb storage."""
+        if self._current_container:
+            container_client = self._database_client.get_container_client(
+                self._current_container
+            )
+            item_names = [item["id"] for item in container_client.read_all_items()]
+            return key in item_names
+        return False
+
+    async def delete(self, key: str) -> None:
+        """Delete the given file from the cosmosdb storage."""
+        if self._current_container:
+            container_client = self._database_client.get_container_client(
+                self._current_container
+            )
+            container_client.delete_item(item=key, partition_key=key)
+
+    async def clear(self) -> None:
+        """Clear the cosmosdb storage."""
+
+    def keys(self) -> list[str]:
+        """Return the keys in the storage."""
+        msg = "CosmosDB storage does yet not support listing keys."
+        raise NotImplementedError(msg)
+
+    def child(self, name: str | None) -> PipelineStorage:
+        """Create a child storage instance."""
+        return self
+
+    def _create_container(self) -> None:
+        """Create a container for the current container name if it doesn't exist."""
+        if self._current_container:
+            partition_key = PartitionKey(path="/id", kind="Hash")
+            self._database_client.create_container_if_not_exists(
+                id=self._current_container,
+                partition_key=partition_key,
+            )
+
+    def _delete_container(self) -> None:
```

Comment:
Same comment as above. This method is never called and is not required by the abstract class. Should we just delete it?